sympy/functions/elementary/hyperbolic.py

```diff
@@ -273,6 +278,53 @@ class cosh(HyperbolicFunction):
     sinh, tanh, acosh
     """
 
+    def _eval_is_positive(self):
+        arg = self.args[0]
+
+        if arg.is_real:
+            return True
+
+        re, im = arg.as_real_imag()
+        im_mod = im % (2*pi)
+
+        if im_mod == 0:
+            return True
+
+        if re == 0:
+            if im_mod < pi/2 or im_mod > 3*pi/2:
+                return True
+            elif im_mod >= pi/2 or im_mod <= 3*pi/2:
+                return False
+
+        return fuzzy_or([fuzzy_and([fuzzy_bool(Eq(re, 0)),
+                         fuzzy_or([fuzzy_bool(im_mod < pi/2),
+                                   fuzzy_bool(im_mod > 3*pi/2)])]),
+                         fuzzy_bool(Eq(im_mod, 0))])
+
+
+    def _eval_is_nonnegative(self):
+        arg = self.args[0]
+
+        if arg.is_real:
+            return True
+
+        re, im = arg.as_real_imag()
+        im_mod = im % (2*pi)
+
+        if im_mod == 0:
+            return True
+
+        if re == 0:
+            if im_mod <= pi/2 or im_mod >= 3*pi/2:
+                return True
+            elif im_mod > pi/2 or im_mod < 3*pi/2:
+                return False
```

Comment:
This raises if the inequalities are unable to evaluate:
```
>>> from sympy import cosh, I, Symbol
>>> x = Symbol('x', real=True)
>>> cosh(I*x).is_positive
>>> cosh(I*x, evaluate=False).is_positive
Traceback (most recent call last):
  File "/Users/enojb/current/sympy/sympy/sympy/core/assumptions.py", line 262, in getit
    return self._assumptions[fact]
KeyError: 'positive'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/enojb/current/sympy/sympy/sympy/core/assumptions.py", line 266, in getit
    return _ask(fact, self)
  File "/Users/enojb/current/sympy/sympy/sympy/core/assumptions.py", line 309, in _ask
    a = evaluate(obj)
  File "/Users/enojb/current/sympy/sympy/sympy/functions/elementary/hyperbolic.py", line 292, in _eval_is_positive
    if im_mod == 0:
  File "/Users/enojb/current/sympy/sympy/sympy/core/relational.py", line 376, in __nonzero__
    raise TypeError("cannot determine truth value of Relational")
TypeError: cannot determine truth value of Relational
```

Reply:
I will add a test for this in the follow up PR(already opened) shortly. Thanks for this. Probably, a `fuzzy_bool` will correct these errors.